### PR TITLE
Align behavior of external PRs

### LINKED ISSUE #107: Align behavior of external PRs
We need to test, if this action has enough permissions to work with the new integration. As we almost exclusively only read data about PR, the token should have enough permissions for most of the actions. Others (like requesting review) needs to be verified.

Review all workflows and decide where it makes sense to add/remove this
```
    # PRs from forks don't have required token authorization
    if: |
        github.event.pull_request.head.repo.full_name == github.repository
```

### PR DESCRIPTION
**Pull request by sebastien-marichal**
Fixes #107

### CODE REVIEW

File: lib/OctokitAction.ts [hunk 1]
```diff
@@ -108,7 +102,7 @@ export abstract class OctokitAction extends Action {
                   }
               }
           }`);
-    return nodes;
+    return data?.organization?.samlIdentityProvider?.externalIdentities?.nodes ?? [];
```
Comment (sebastien-marichal) [inline lib/OctokitAction.ts:105]: With destructing, if any property is null, we can get an error like:
```
TypeError: Cannot read properties of null (reading 'externalIdentities')
```
Which will stop the execution of the action.

Comment (pavel-mikula-sonarsource) [reply lib/OctokitAction.ts:105]: It's probably better to let it fail hard. Otherwise token issues would be hidden deep under indirect indices

**Review by pavel-mikula-sonarsource — CHANGES_REQUESTED** [review]
Please revert the GraphQL change

**Review by pavel-mikula-sonarsource — APPROVED** [review]
LGTM

**Pull request merged**